includes/class-wc-post-types.php

```diff
@@ -312,6 +312,7 @@ public static function register_post_types() {
 					'has_archive'         => $has_archive,
 					'show_in_nav_menus'   => true,
 					'show_in_rest'        => true,
+					'menu_icon'           => 'data:image/svg+xml;base64,' . base64_encode( str_replace( '<svg', '<svg fill="black"', get_gridicon( 'gridicons-product' ) ) ), // SVG 'fill' attribute must be set for menu colors to apply correctly.
```

Comment:
The `str_replace` here doesn't feel great, but the icon has to have a defined `fill` attribute for the admin menu default colors to apply correctly. It's not pretty, but this does the job.